### DISCUSSION TITLE
Where is the image compression code?

### DISCUSSION THREAD

**Discussion by orazioedoardo** [AsamK/signal-cli#928] [Q&A]
I've noticed that `signal-cli` compresses image attachments like the official client, but I can't find the code that does this in the repo.

**Comment by AsamK** [comment]
No, signal-cli doesn't compress images.
When sending an attachment it uploads the file content as is.

**Reply by orazioedoardo** [reply]
Looks like it does, I've sent myself this picture from [wikimedia](https://upload.wikimedia.org/wikipedia/commons/2/2b/NYC_Downtown_Manhattan_Skyline_seen_from_Paulus_Hook_2019-12-20_IMG_7347_FRD_%28cropped%29.jpg), SHA1 `afcc753d1b074975aa5910a8d409574e424f17da`, 3869 × 1551 pixel. The received image is 3072 × 1231 pixel and obviously has a different checksum. Original metadata is also stripped. New metadata shows `ProfileCopyright` set to `Google Inc. 2016`.

**Reply by orazioedoardo** [reply]
Sorry to bother @AsamK, have you tried to reproduce this?

**Reply by exquo** [reply]
I think it's the signal servers that do the resizing and metadata stripping.

**Reply by orazioedoardo** [reply]
How would the server do that when the attachment is encrypted?

**Reply by exquo** [reply]
You're right, it should be encrypted on the device. I've been using too much Telegram lately, so had been thinking in that framework..
For reference, here's the  signal-android's implementation: https://github.com/signalapp/Signal-Android/pull/7542.
